susemanager-utils/susemanager-sls/salt/bootstrap/init.sls

```diff
@@ -127,16 +129,39 @@ bootstrap_repo:
       - ([ {{ bootstrap_repo_exists }} = "True" ])
 {%- endif %}
 
+{%- if grains['os_family'] == 'RedHat' %}
+trust_res_gpg_key:
+  cmd.run:
+    - name: rpm --import https://{{ salt['pillar.get']('mgr_server') }}/pub/{{ salt['pillar.get']('gpgkeys:res:file') }}
+    - unless: rpm -q {{ salt['pillar.get']('gpgkeys:res:name') }}
+    - runas: root
+{%- elif grains['os_family'] == 'Debian' %}
+install_gnupg_debian:
+  pkg.latest:
+    - pkgs:
+      - gnupg
+{%- endif %}
+
 {% include 'channels/gpg-keys.sls' %}
 
+{%- set salt_minion_name = 'salt-minion' %}
+{%- set salt_config_dir = '/etc/salt' %}
+{% set venv_available_request = salt['http.query'](bootstrap_repo_url + 'venv-enabled-' + grains['osarch'] + '.txt', status=True, verify_ssl=False) %}
+{# Prefer venv-salt-minion if available and not disabled #}
+{%- set use_venv_salt = (0 < venv_available_request.get('status', 404) < 300) and not salt['pillar.get']('mgr_avoid_venv_salt_minion') %}
+{%- if use_venv_salt %}
+{%- set salt_minion_name = 'venv-salt-minion' %}
+{%- set salt_config_dir = '/etc/opt/venv-salt-minion' %}
+{%- endif -%}
+
 salt-minion-package:
   pkg.installed:
-    - name: salt-minion
+    - name: {{ salt_minion_name }}
     - install_recommends: False
     - require:
       - file: bootstrap_repo
 
-/etc/salt/minion.d/susemanager.conf:
+{{ salt_config_dir }}/minion.d/susemanager.conf:
```

Comment:
nitpick

I am not a fan of filenames as "ids". As we touch them anyway, would it make sense to specify them as "name" and add a real id (mgr_something) to all of these rules?

Reply:
in case of using ids for each file we will have 4 vars instead of one and probably make sls less readable as have to check what each id should contain, for this case `salt_config_dir` defines the root of salt config and we see the relative path of the file. It's looks more clear for me.

Reply:
What I mean is:
```
mgr_susemanager_conf:
  file.managed:
    - name: {{ salt_config_dir }}/minion.d/susemanager.conf
    - source:
      - salt://bootstrap/susemanager.conf
   - template: jinja
   - mode: 644
    - require:
      - pkg: salt-minion-package
```

Why do we need 4 vars for this change?
Of cause all the states below changed in the same way and the require statements need to name the id instead of the file path.
But as I said, this is optional

Reply:
There are few more files handled inside `salt_config_dir` the same way in case of using distinct id to exact this file it's better to use the same approach for all of the files and there are 4 files like this I found on the first glance